src/shell/shell.cpp

```diff
@@ -452,6 +453,49 @@ void DOS_Shell::SyntaxError()
 	WriteOut(MSG_Get("SHELL_SYNTAX_ERROR"));
 }
 
+static std_fs::path get_cmd_history_path()
+{
+	return get_platform_config_dir() / "cmd_history.txt";
+}
+
+void DOS_Shell::ReadCommandHistory()
+{
+	std::ifstream history_file(get_cmd_history_path());
+	if (history_file) {
+		std::string line;
+		while (getline(history_file, line)) {
+			if (!line.empty()) {
+				history.push_back(line);
+			}
+		}
+	}
+}
+
+void DOS_Shell::WriteCommandHistory()
+{
+	std_fs::path history_path = get_cmd_history_path();
+	std::ofstream history_file(history_path);
+	if (!history_file) {
+		LOG_WARNING("SHELL: Unable to write command history to file: %s", history_path.string().c_str());
```

Comment:
Enclose filenames in single quotes in log messages.

Reply:
Fixed